### ISSUE TITLE
Text in type=number is not registered at all

### ISSUE THREAD

**Issue by MarKayS** [Dogfalo/materialize#1371]
Hi, I just started using materialize and apart of few problems its great! But when i use any input of type number and add class validate, everything works fine unless i input some text (non digits) in it. 

In that case input behaves like it would be empty, label will overlap the typed text and validate won't trigger. I've tried to make my own validation, but can't find any suitable way since the .val() is empty.

Any help? Thanks

http://codepen.io/anon/pen/doXPPR

![image](https://cloud.githubusercontent.com/assets/7431147/7662465/0501f7fe-fb5d-11e4-875e-e3fd77155cf9.png)

**Comment by Dogfalo** [comment]
You can check the validity state of it. This is a fix we will implement in the coming weeks

**Comment by MarKayS** [comment]
Thanks for the quick answer. I found that if I set val to empty string myself the text would disappear, which is kind of weird because if I just try to find out the val() its empty on its own. I was trying to do to make it recognize the text and then change class to invalid, but that's too much for my beginner's jquery skills. I will check after some while for new version after the issue has been resolved.

Thanks again, keep up the excellent work.
